src/cpu_profile.cc

```diff
@@ -12,93 +12,95 @@ namespace nodex {
   using v8::Object;
   using v8::ObjectTemplate;
   using v8::FunctionTemplate;
-  using v8::Persistent;
   using v8::String;
   using v8::Function;
   using v8::Value;
 
-  Persistent<ObjectTemplate> Profile::profile_template_;
-  Persistent<Array> Profile::profiles;
+  Nan::Persistent<ObjectTemplate> Profile::profile_template_;
+  Nan::Persistent<Array> Profile::profiles;
   uint32_t Profile::uid_counter = 0;
-  
+
+  NAN_METHOD(Profile_EmptyMethod) {
+  }
+
   void Profile::Initialize () {
-    NanScope();
-    
-    Local<FunctionTemplate> f = NanNew<FunctionTemplate>();
+    Nan::HandleScope scope;
+
+    Local<FunctionTemplate> f = Nan::New<FunctionTemplate>(Profile_EmptyMethod);
     Local<ObjectTemplate> o = f->InstanceTemplate();
     o->SetInternalFieldCount(1);
-    NODE_SET_METHOD(o, "delete", Profile::Delete);
-    NanAssignPersistent(profile_template_, o);
+    Nan::SetMethod(o, "delete", Profile::Delete);
+    profile_template_.Reset(o);
   }
 
   NAN_METHOD(Profile::Delete) {
-    NanScope();
-    
-    Handle<Object> self = args.This();
-    void* ptr = NanGetInternalFieldPointer(self, 0);
-    Local<Array> profiles = NanNew<Array>(Profile::profiles);
-    
+    Local<Object> self = info.This();
+    void* ptr = Nan::GetInternalFieldPointer(self, 0);
+    Local<Array> profiles = Nan::New<Array>(Profile::profiles);
+
     uint32_t count = profiles->Length();
     for (uint32_t index = 0; index < count; index++) {
-      if (profiles->Get(index) == args.This()) {
+      if (profiles->Get(index) == info.This()) {
         Local<Value> argv[2] = {
-          NanNew<Integer>(index),
-          NanNew<Integer>(1)
+          Nan::New<Integer>(index),
+          Nan::New<Integer>(1)
         };
-        Handle<Function>::Cast(profiles->Get(NanNew<String>("splice")))->Call(profiles, 2, argv);
+        Local<Function>::Cast(profiles->Get(Nan::New<String>("splice").ToLocalChecked()))->Call(profiles, 2, argv);
         break;
       }
     }
     static_cast<CpuProfile*>(ptr)->Delete();
-
-    NanReturnUndefined();
   }
 
-  Handle<Value> Profile::New (const CpuProfile* node) {
-    NanEscapableScope();
-    
+  Local<Value> Profile::New (const CpuProfile* node) {
+    Nan::EscapableHandleScope scope;
+
     if (profile_template_.IsEmpty()) {
       Profile::Initialize();
     }
-    
+
     uid_counter++;
-      
-    Local<Object> profile = NanNew(profile_template_)->NewInstance();
-    NanSetInternalFieldPointer(profile, 0, const_cast<CpuProfile*>(node));
-    
-    Local<Value> CPU = NanNew<String>("CPU");
-    Local<Value> uid = NanNew<Integer>(uid_counter);
-    Handle<String> title = node->GetTitle();
+
+    Local<Object> profile = Nan::New(profile_template_)->NewInstance();
+    Nan::SetInternalFieldPointer(profile, 0, const_cast<CpuProfile*>(node));
+
+    Local<Value> CPU = Nan::New<String>("CPU").ToLocalChecked();
+    Local<Value> uid = Nan::New<Integer>(uid_counter);
+#if (NODE_MODULE_VERSION >= 45)
```

Comment:
I think this can be considered part of the exotic API that no-one but node-inspector uses, so I don't think it's worth changing anything in NAN. Handles should not be used in general. Just do an ifdef and v8::Local::New() the handle.